lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala

```diff
@@ -92,6 +100,38 @@ class GradleBuildTool(index: IndexCommand) extends BuildTool("Gradle", index) {
           |      $executable
           |    }
           |  }
+          |  task $lsifJavaDependencies {
+          |    def depsOut = java.nio.file.Paths.get('$dependenciesPath')
+          |    doLast {
+          |      tasks.withType(JavaCompile) {
+          |        try {
+          |          configurations.each { config ->
+          |              def artifactType = org.gradle.api.attributes.Attribute.of("artifactType", String.class)
+          |              def attributeType = "jar"
+          |              if (config.canBeResolved) {
+          |                def artifacts = config.incoming.artifactView { view ->
+          |                  view.lenient = true
+          |                  view.attributes { container ->
+          |                    container.attribute(artifactType, attributeType)
+          |                  }
+          |                }.artifacts
```

Comment:
Really interesting! How did you come across this solution?

Reply:
I adapted from the Bloop plugin https://github.com/scalacenter/bloop/blob/4d9f966b67ff726855cb36ec3565a34d00bf41ec/integrations/gradle-bloop/src/main/scala/bloop/integrations/gradle/model/BloopConverter.scala#L277-L299

Reply:
Damn, wish I knew of this many months ago :laughing: